.github/workflows/run-benchmark.yml

```diff
@@ -0,0 +1,163 @@
+
+    env:
+      PR_COMMENT: null # will be populated later
+
+    steps:
+      - name: Configure RunsOn
+        uses: runs-on/action@v1
+
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure git
+        run: git config --global --add safe.directory "${GITHUB_WORKSPACE}"
+
+      # jemalloc is a dependency of the Benchmarking package
+      # actions/cache will detect zstd and will become much faster.
+      - name: Install jemalloc, curl, jq and zstd
+        run: |
+          apt-get update -y
+          apt-get install -y libjemalloc-dev curl jq zstd
+
+      - name: Restore .build
+        id: restore-cache
+        uses: actions/cache/restore@v4
+        with:
+          path: Benchmarks/.build
+          key: "swiftpm-benchmark-build-${{ runner.os }}-${{ github.event.pull_request.base.sha || github.event.after }}"
+          restore-keys: "swiftpm-benchmark-build-${{ runner.os }}-"
+
+      - name: Run benchmarks for branch '${{ github.head_ref || github.ref_name }}'
+        run: |
+          swift package -c release --disable-sandbox \
+            --package-path Benchmarks \
+            benchmark baseline update \
+            '${{ github.head_ref || github.ref_name }}'
+
+      - name: Read benchmark result
+        id: read-benchmark
+        run: |
+          swift package -c release --disable-sandbox \
+            --package-path Benchmarks \
+            benchmark baseline read \
+            '${{ github.head_ref || github.ref_name }}' \
+            --no-progress \
+            --format markdown \
+            >> result.text
+
+          # Read the result to the output of the step
+          echo 'result<<EOF' >> "${GITHUB_OUTPUT}"
+          cat result.text >> "${GITHUB_OUTPUT}"
+          echo 'EOF' >> "${GITHUB_OUTPUT}"
+
+      - name: Compare branch '${{ github.head_ref || github.ref_name }}' against thresholds
+        id: compare-benchmark
+        run: |
+          TIMESTAMP=$(date -u +"%Y-%m-%d %H:%M:%S UTC")
+          ENCODED_TIMESTAMP=$(date -u +"%Y-%m-%dT%H%%3A%M%%3A%SZ")
+          TIMESTAMP_LINK="https://www.timeanddate.com/worldclock/fixedtime.html?iso=$ENCODED_TIMESTAMP"
+          echo "## Benchmark check running at [$TIMESTAMP]($TIMESTAMP_LINK)" >> summary.text
+
+          # Disable 'set -e' to prevent the script from exiting on non-zero exit codes
+          set +e
+          swift package -c release --disable-sandbox \
+            --package-path Benchmarks \
+            benchmark thresholds check \
+            '${{ github.head_ref || github.ref_name }}' \
+            --path "$PWD/Benchmarks/Thresholds/" \
+            --no-progress \
+            --format markdown \
+            >> summary.text
+          echo "exit-status=$?" >> "${GITHUB_OUTPUT}"
+          set -e
+
+          echo 'summary<<EOF' >> "${GITHUB_OUTPUT}"
+          cat summary.text >> "${GITHUB_OUTPUT}"
+          echo 'EOF' >> "${GITHUB_OUTPUT}"
```

Comment:
Again, why not just read the file at point of use?

Reply:
see the other comment